src/pager.c

```diff
@@ -1552,14 +1570,17 @@ char *supplemental_name;
 
             /* monster lookup: try to parse as a monster */
             pm = NULL;
-            int mndx = name_to_mon(dbase_str);
-            if (mndx != NON_PM) {
+            /* first try with material, then try base version if no results */
+            int mndx;
+            if ((mndx = name_to_mon(material)) == NON_PM)
+                mndx = name_to_mon(dbase_str);
```

Comment:
In the absence of a monster materials patch, this can just check the material version of the string instead of dbase_str. If the two strings differ, the one containing the material is correct.